### PR TITLE
Update to 1.4 (via Axeel)

### PR DESCRIPTION
**Pull request by hakusaro**
This is a PR for Axeel's 1.4 branch into `general-devel`.

### CODE REVIEW

**Comment by AxeelAnder** [comment]
this branch is from master and I deleted upcoming features in changelog bacause I thought master branch is for release so there shouldn't be upcoming features.

So pls don't approve this

**Comment by hakusaro** [comment]
hey wait, I wanted this

File: TShockAPI/Commands.cs [hunk 1]
```diff
@@ -2144,7 +2145,10 @@ private static void ClearAnglerQuests(CommandArgs args)
 
 		private static void ToggleExpert(CommandArgs args)
 		{
-			Main.expertMode = !Main.expertMode;
+			if (Main.GameMode != GameModeID.Expert)
```
Comment (hakusaro) [inline TShockAPI/Commands.cs:2148]: This is nice.

File: TShockAPI/Commands.cs [hunk 2]
```diff
@@ -2274,7 +2278,7 @@ private static void SpawnBoss(CommandArgs args)
 					return;
 				case "wof":
 				case "wall of flesh":
-					if (Main.wof >= 0)
+					if (Main.wofNPCIndex != -1)
```
Comment (hakusaro) [inline TShockAPI/Commands.cs:2281]: Note: This is a delta between @Patrikkk's changes: `Main.wofNPCIndex >= 0`

File: TShockAPI/DB/CharacterManager.cs [hunk 3]
```diff
@@ -175,7 +175,7 @@ public bool InsertPlayerData(TSPlayer player, bool fromCommand = false)
...
+						player.Account.ID, playerData.health, playerData.maxHealth, playerData.mana, playerData.maxMana, String.Join("~", playerData.inventory), playerData.extraSlot, player.TPlayer.SpawnX, player.TPlayer.SpawnY, player.TPlayer.skinVariant, player.TPlayer.hair, player.TPlayer.hairDye, TShock.Utils.EncodeColor(player.TPlayer.hairColor), TShock.Utils.EncodeColor(player.TPlayer.pantsColor),TShock.Utils.EncodeColor(player.TPlayer.shirtColor), TShock.Utils.EncodeColor(player.TPlayer.underShirtColor), TShock.Utils.EncodeColor(player.TPlayer.shoeColor), TShock.Utils.EncodeBoolArray(player.TPlayer.hideVisibleAccessory), TShock.Utils.EncodeColor(player.TPlayer.skinColor),TShock.Utils.EncodeColor(player.TPlayer.eyeColor), player.TPlayer.anglerQuestsFinished);
```
Comment (hakusaro) [inline TShockAPI/DB/CharacterManager.cs:178]: This is good.

File: TShockAPI/NetItem.cs [hunk 4]
```diff
@@ -16,7 +16,7 @@
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-﻿using System;
+ using System;
```
Comment (hakusaro) [inline TShockAPI/NetItem.cs:19]: Typo

**Review by hakusaro — COMMENTED** [review]
looks good to me, will edit changelog and update

**Pull request merged**